[webview_flutter] Added missing settings when developing initial AndroidWebViewController

mvanbeusekom opened the pull request:
Added the following setters which I missed during initial development:

- `setBuiltInZoomControls`
- `setDisplayZoomControls`
- `setJavaScriptCanOpenWindowsAutomatically`
- `setLoadWithOverviewMode`
- `setSupportMultipleWindows`
- `setUseWideViewPort`

Part of issue https://github.com/flutter/flutter/issues/94051

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

bparrishMines commented on packages/webview_flutter/webview_flutter_android/lib/src/v4/src/android_webview_controller.dart:
```diff
@@ -266,14 +303,79 @@ class AndroidWebViewController extends PlatformWebViewController {
       _webView.settings.setUserAgentString(userAgent);
+
+  /// Sets the restrictions that apply on automatic media playback.
+  Future<void> setAutoMediaPlaybackPolicy(
```
Since this is Android only, I think this would be better as `setMediaPlaybackRequiresUserGesture(bool)`. The enum was used before because iOS could set the specific media that requires gestures.

mvanbeusekom replied:
Updated the method callback to match the Android specific name.

bparrishMines reviewed (commented):
I don't think we should expose support for the following methods yet.

```
setBuiltInZoomControls
setDisplayZoomControls
setJavaScriptCanOpenWindowsAutomatically
setLoadWithOverviewMode
setSupportMultipleWindows
setUseWideViewPort
```

All of these are called on the WebView in the constructor because they are required to display the `WebView` correctly as a PlatformView.

These should just be set in the constructor to the same values we have in the current implementation:

https://github.com/flutter/plugins/blob/main/packages/webview_flutter/webview_flutter_android/lib/webview_android_widget.dart#L119

mvanbeusekom commented:
> I don't think we should expose support for the following methods yet.
> 
> ```
> setBuiltInZoomControls
> setDisplayZoomControls
> setJavaScriptCanOpenWindowsAutomatically
> setLoadWithOverviewMode
> setSupportMultipleWindows
> setUseWideViewPort
> ```
> 
> All of these are called on the WebView in the constructor because they are required to display the `WebView` correctly as a PlatformView.
> 
> These should just be set in the constructor to the same values we have in the current implementation:
> 
> https://github.com/flutter/plugins/blob/main/packages/webview_flutter/webview_flutter_android/lib/webview_android_widget.dart#L119

Updated the `AndroidWebViewController` to initialise the `android_view.webView` with default settings in it's constructor. Removed the separate setters.

bparrishMines commented on packages/webview_flutter/webview_flutter_android/lib/src/v4/src/android_webview_controller.dart:
```diff
@@ -258,6 +266,15 @@ class AndroidWebViewController extends PlatformWebViewController {
+  /// default value is true for apps targeting Build.VERSION_CODES.Q and below,
+  /// and false when targeting Build.VERSION_CODES.R and above.
+  Future<void> enableFileAccess(bool enabled) =>
+      _webView.settings.setAllowFileAccess(enabled);
```
Ah, I forgot about this one. This is already called in the `loadFile` method. `enableFileAccess` can be removed.

bparrishMines commented:
This can be closed since this has already been implemented: https://github.com/flutter/plugins/blob/main/packages/webview_flutter/webview_flutter_android/lib/src/android_webview_controller.dart#L72
